components/header/SearchBar.tsx

```diff
@@ -1,40 +1,180 @@
+import axios from "axios";
+import { useNavigation, NavigationProp } from "@react-navigation/native";
+import { useDispatch } from "react-redux";
+import { setCity } from "../../slices/citySlice";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { RootStackParamList } from "../../types/navigation";
+const BASE_URL_GEONAMES = "https://secure.geonames.org/searchJSON";
+
+interface GeoName {
+  geonameId: number;
+  name: string;
+  countryName: string;
+}
 
 const SearchBar = () => {
+  const [query, setQuery] = useState("");
+  const [suggestions, setSuggestions] = useState<GeoName[]>([]);
   const { styles } = useStyles(stylesheet);
+  const navigation: NavigationProp<RootStackParamList> = useNavigation();
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  const debounce = (func: (text: string) => Promise<void>, delay: number) => {
+    let timeoutId: ReturnType<typeof setTimeout>;
+
+    return (...args: [string]) => {
+      clearTimeout(timeoutId);
+
+      timeoutId = setTimeout(() => {
+        func(...args);
+      }, delay);
+    };
+  };
+
+  const fetchCities = async (text: string) => {
+    if (text.length > 2) {
+      try {
+        const response = await axios.get(
+          `${BASE_URL_GEONAMES}?name_startsWith=${text}&maxRows=5&username=${process.env.EXPO_PUBLIC_USERNAME}`
+        );
+        setSuggestions(response.data.geonames);
+      } catch (error) {
+        console.error("Error fetching cities:", JSON.stringify(error, null, 2));
+        setSuggestions([]);
+      } finally {
+        setLoading(false);
+      }
+    } else {
+      setSuggestions([]);
+      setLoading(false);
+    }
+  };
+
+  const debouncedFetchCities = debounce(fetchCities, 400);
+
+  const handleSearch = (text: string) => {
+    setLoading(true);
+    setQuery(text);
+    debouncedFetchCities(text);
+  };
+
+  const handleCitySelection = async (cityName: string) => {
+    setQuery(cityName);
+    setSuggestions([]);
+    dispatch(setCity(cityName));
+    try {
+      await AsyncStorage.setItem("selectedCity", cityName);
+    } catch (error) {
+      console.error(error);
+    }
+    navigation.navigate("Tabs");
+  };
 
   return (
-    <View style={styles.searchBarContainer}>
-      <TextInput style={styles.searchBar} placeholder="Search city..." />
-      <TouchableOpacity style={styles.searchButton} onPress={() => {}}>
-        <Ionicons name="search" size={20} color={styles.searchButton.color} />
-      </TouchableOpacity>
-    </View>
+    <>
+      <View style={styles.container}>
+        <View style={styles.searchBarContainer}>
+          <TextInput
+            style={styles.searchBar}
+            placeholder="Search city..."
+            value={query}
+            onChangeText={(text) => handleSearch(text)}
+          />
+          <TouchableOpacity style={styles.searchButton} onPress={() => {}}>
+            <Ionicons name="search" size={20} color="#2C3E50" />
```

Comment:
Ten kolorek mamy już zdefiniowany w theme'ie

https://reactnativeunistyles.vercel.app/reference/theming/#get-the-current-theme